pkg/utils/sharedcache/sharedcache.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sharedcache
```

Comment:
Why not just inject this into the controller at initialization time like we do with our other caches?

Reply:
Something like [this](https://github.com/BEvgeniyS/karpenter/pull/2/files)?

I've tried it at first but the amount of changes seemed too much. What do you think, what's better?

Reply:
It's how we do things today, in the codebase. We wire dependencies around explicitly, rather than global singletons or a DI framework. 

Usually you can just instantiate it as part of the parent class's constructor. If used in multiple places, we wire it up as needed.

Reply:
Understood. I definitely want to follow existing practices.

Can you review https://github.com/BEvgeniyS/karpenter/pull/2/files then? I'll just push the changes to this branch if it's correct

Reply:
updated this PR to use injection